chore: always show the archived conversations [WPB-7307]

caldrian opened the pull request:
<!--do not remove this marker, its needed to replace info when ticket title is updated -->
<!--jira-description-action-hidden-marker-start-->

<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-7307" title="WPB-7307" target="_blank"><img alt="Story" src="https://wearezeta.atlassian.net/rest/api/2/universal_avatar/view/type/issuetype/avatar/10815?size=medium" />WPB-7307</a>  [iOS] : Remove "Contacts" "Folder"
  </td></table>
  <br />
 

<!--jira-description-action-hidden-marker-end-->
<!--do not remove this marker, its needed to replace info when ticket title is updated -->

<!--do not remove this marker, its needed to replace info when ticket title is updated -->

<!--do not remove this marker, its needed to replace info when ticket title is updated -->

<!--do not remove this marker, its needed to replace info when ticket title is updated -->

<!--do not remove this marker, its needed to replace info when ticket title is updated -->

<!--do not remove the jira markers to link tickets automatically -->


### Issue

According to the designer, there should always be at least three tabs in the tab bar.
As long as we don't have any other tabs to show (folders will be removed in a subsequent ticket/PR), we always keep the archived conversations visible.

---

### Checklist

- [x] Title contains a reference JIRA issue number like `[WPB-XXX]`.
- [x] Description is filled and free of optional paragraphs.
- [x] Adds/updates automated tests.

caldrian committed:
chore: always show the archived conversations tab

caldrian committed:
update snapshot test reference images

caldrian committed:
remove unused code

caldrian committed:
Merge branch 'develop' of github.com:wireapp/wire-ios into chore/always-show-archived-conversations-WPB-7307

# Conflicts:
#	wire-ios/Wire-iOS Tests/ConversationList/Container/MockConversationListContainer.swift
#	wire-ios/Wire-iOS Tests/ConversationList/ConversationListViewModel/ConversationListViewModelTests.swift
#	wire-ios/Wire-iOS Tests/ConversationListTabBarTests.swift

agisilaos reviewed (approved):
Please fix the SwiftLint warnings before merging! 🙏 Looking great, thank you!

caldrian committed:
ensure conversation list tab is selected

# Conflicts:
#	wire-ios/Wire-iOS/Sources/UserInterface/ConversationList/Container/ConversationListViewController.swift
#	wire-ios/Wire-iOS/Sources/UserInterface/ConversationList/ConversationListTabBar.swift

caldrian committed:
Merge branch 'epic/navigation-overhaul-WPB-6647' into chore/always-show-archived-conversations-WPB-7307

caldrian committed:
fix SwiftLint warning

caldrian committed:
fix snapshot test reference images
